receiver/prometheusreceiver/metrics_receiver_target_allocator_test.go

```diff
@@ -532,10 +642,23 @@ func TestTargetAllocatorJobRetrieval(t *testing.T) {
 						// which is identical to the source url
 						s.Labels["__meta_url"] = model.LabelValue(sdConfig.URL)
 						require.Equal(t, s.Labels, group.Labels)
+
+						if s.MetricRelabelConfig != nil {
+							// Adding wait here so that the latest scrape config is applied with the updated regex
+							time.Sleep(5 * time.Second)
```

Comment:
Putting a 5s sleep inside a triply-nested loop can have a significant impact on test runtime.  Can this be avoided?

Reply:
@Aneurysm9 - Unfortunately, looks like if the sleep is not present, the updated regex doesnt get picked up. This is the only test that uses the sleep and for all the other tests it wouldn't go in this loop.